### PR TITLE
Initial CI info

### PR DESCRIPTION
**Pull request by nigurr**
* Final PR for telemetry publish

### CODE REVIEW

File: Tasks/VsTest/package.json [hunk 1]
```diff
@@ -17,12 +17,12 @@
   "homepage": "https://github.com/Microsoft/vsts-tasks#readme",
   "dependencies": {
     "performance-now": "0.2.0",
+    "string": "^3.3.1",
+    "vso-node-api": "file:..\\..\\..\\vsts-node-api\\_build",
```
Comment (nigurr) [inline Tasks/VsTest/package.json]: will remove this and update after new vso-node-api is released

File: Tasks/VsTest/cieventlogger.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { ICustomerIntelligenceApi as ciApi } from 'vso-node-api/CustomerIntelligenceApi';
+import { CustomerIntelligenceEvent as ciEvent } from 'vso-node-api/interfaces/CustomerIntelligenceInterfaces';
+
+const area: string = 'TestExecutionTask';
+const feature: string = 'TaskExecution';
+
+const collectionUri = tl.getVariable('System.TeamFoundationCollectionUri');
+const token = tl.getEndpointAuthorization('SystemVssConnection', true).parameters['AccessToken'];
+const webapi: WebApi = new WebApi(collectionUri, getBearerHandler(token));
+const ci: ciApi = webapi.getCustomerIntelligenceApi();
```
Comment (nigurr) [inline Tasks/VsTest/cieventlogger.ts]: remove this line

**Comment by bryanmacfarlane** [comment]
The agent should do this via a best effort ##vso command logger.

**Comment by TingluoHuang** [comment]
Can you paste a sample of the full CI event looks like?
what information will be send out?
```JSON
{
    "key1": "val1",
    "key2": "val2",
}

```

File: Tasks/VsTest/cieventlogger.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import * as tl from 'vsts-task-lib/task';
+
+const area: string = 'TestExecution';
+const feature: string = 'TestExecutionTask';
+
+const collectionUri = tl.getVariable('System.TeamFoundationCollectionUri');
+
+function getDefaultProps() {
```
Comment (TingluoHuang) [inline Tasks/VsTest/cieventlogger.ts]: should agent always do this?

Comment (nigurr) [inline Tasks/VsTest/cieventlogger.ts]: @TingluoHuang we want to attach build/releaseid to trace back the info easily

Comment (nigurr) [inline Tasks/VsTest/cieventlogger.ts]: Sample: 

{
  "uniqueIdentifier": "ae67078a-9340-4e3f-825e-da3f4197770b",
  "anonymousIdentifier": "",
  "hostId": "0efb4611-d565-4cd1-9a64-7d6cb6d7d5f0",
  "identity": "",
  "vsid": "4f8968e9-8ba5-4417-a321-72f07160d96f",
  "cuid": "00000000-0000-0000-0000-000000000000",
  "area": "TestExecution",
  "feature": "TestExecutionTask",
  "timeStamp": "2017-07-20T14:08:06.5533364Z",
  "userAgent": "VSServices/15.255.65000.0 (NetStandard; Microsoft Windows 10.0.15063) VstsAgentCore-win7-x64/2.120.0 (Microsoft Windows 10.0.15063)",
  "properties": {
    "releaseuri": "vstfs:///ReleaseManagement/Release/1340734",
    "releaseid": "1340734",
    "collectionuri": "https://mseng.visualstudio.com/",
    "state": "completed",
    "result": {}
  }
}

**Pull request merged**